### PR TITLE
Expression functions which return file info

### PR DESCRIPTION
**Pull request by nyalldawson**
base_file_name: Returns the base name of the file without the directory or file suffix.
file_exists: Returns true if a file exists
file_name: Returns the file name from a full path
file_path: Returns the directory/path from a full file path
file_size: Returns a file size
file_suffix: Returns a files suffix/extension
is_directory: Returns true if a file path is a directory
is_file: Returns true if a file path is a file

Sponsored by North Road, thanks to SLYR

### CODE REVIEW

**Comment by nirvn** [comment]
Code looks good, merge at will.

**Review by m-kuhn — APPROVED** [review]
:tada:

File: resources/function_help/json/file_name [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "file_name",
+  "type": "function",
+  "description": "Returns the name of a file (including the file extension), excluding the directory.",
+  "arguments": [ {"arg":"path","description":"a file path"}],
+  "examples": [ { "expression":"base_file_name('/home/qgis/data/country_boundaries.shp')", "returns":"'country_boundaries.shp'"}]
```
Comment (Gustry) [inline resources/function_help/json/file_name]: you forgot to update `base_file_name` example

File: resources/function_help/json/file_path [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "file_path",
+  "type": "function",
+  "description": "Returns the directory component of a file path. This does not include the file name.",
+  "arguments": [ {"arg":"path","description":"a file path"}],
+  "examples": [ { "expression":"base_file_name('/home/qgis/data/country_boundaries.shp')", "returns":"'/home/qgis/data'"}]
```
Comment (Gustry) [inline resources/function_help/json/file_path]: same

File: resources/function_help/json/file_suffix [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "file_suffix",
+  "type": "function",
+  "description": "Returns the file suffix (extension) from a file path.",
+  "arguments": [ {"arg":"path","description":"a file path"}],
+  "examples": [ { "expression":"base_file_name('/home/qgis/data/country_boundaries.shp')", "returns":"'shp'"}]
```
Comment (Gustry) [inline resources/function_help/json/file_suffix]: same

**Review by Gustry — COMMENTED** [review]
Nice addition!

**Comment by andreasneumann** [comment]
Nice - it will be useful! Thanks!

File: resources/function_help/json/file_suffix (hunk 3 above)
Comment (nyalldawson) [reply resources/function_help/json/file_suffix]: Thanks, fixed all those!

**Comment by 3nids** [comment]
would that hurt to backport to 3.4 or is this really seen as a feature (no offense ;) )

**Pull request merged**